modules/postprocessing/src/processors/depthoffield.cpp

```diff
@@ -0,0 +1,427 @@
+
+    if (approximate_) {
+        // Prepare light field
+        dimLightField_ = size3_t(dim.x, dim.y, viewCountApprox_.get() + simViewCountApprox_.get());
+        lightField_ = std::make_shared<Volume>(dimLightField_, DataFormat<vec4>::get());
+        lightFieldDepth_ = std::make_shared<Volume>(dimLightField_, DataFormat<float>::get());
+        float* lightFieldDepthData = static_cast<float*>(
+            lightFieldDepth_->getEditableRepresentation<VolumeRAM>()->getData());
+        std::fill(lightFieldDepthData,
+                  lightFieldDepthData + dimLightField_.x * dimLightField_.y * dimLightField_.z,
+                  -1.f);
+
+        // Prepare Halton sequence
+        haltonX_ = util::haltonSequence<float>(2, simViewCountApprox_.get());
+        haltonImg_ = std::make_shared<Image>(size2_t(simViewCountApprox_.get(), 1),
+                                             DataFormat<float>::get());
+        float* haltonData = static_cast<float*>(
+            haltonImg_->getColorLayer()->getEditableRepresentation<LayerRAM>()->getData());
+        std::copy(haltonX_.begin(), haltonX_.end(), haltonData);
+    } else {
+        // Prepare accumulation buffer
+        prevOutImg_ = std::make_shared<Image>(dim, DataFormat<vec4>::get());
+        nextOutImg_ = std::make_shared<Image>(dim, DataFormat<vec4>::get());
+
+        // Prepare Halton sequences
+        haltonX_ = util::haltonSequence<float>(2, maxEvalCount);
+        haltonY_ = util::haltonSequence<float>(3, maxEvalCount);
+    }
+
+    // Update focus depth and aperture ranges based on the depths in the scene, ignoring
+    // background points and including the current value.
+    const float* inputDepthData = static_cast<const float*>(
+        img->getRepresentation<ImageRAM>()->getDepthLayerRAM()->getData());
+    float minDepthNdc = 1;
+    float maxDepthNdc = 0;
+    for (const float* f = inputDepthData; f < inputDepthData + dim.x * dim.y; f++) {
+        minDepthNdc = std::min(minDepthNdc, *f);
+        if (*f < 1) {
+            maxDepthNdc = std::max(maxDepthNdc, *f);
+        }
+    }
+
+    double nearClip = camera_.getNearPlaneDist();
+    double farClip = camera_.getFarPlaneDist();
+    float minDepthWorld = nearClip * farClip / (minDepthNdc * (nearClip - farClip) + farClip);
+    float maxDepthWorld = nearClip * farClip / (maxDepthNdc * (nearClip - farClip) + farClip);
+
+    focusDepth_.setMinValue(std::min(minDepthWorld, focusDepth_.get()));
+    focusDepth_.setMaxValue(std::max(maxDepthWorld, focusDepth_.get()));
+    focusDepth_.set(focusDepth_.get(), std::min(minDepthWorld, focusDepth_.get()),
+                    std::max(maxDepthWorld, focusDepth_.get()),
+                    (maxDepthWorld - minDepthWorld) / 100.0);
+
+    int depthScale = floor(log10(minDepthWorld));
+    float minAperture = pow(10, depthScale - 1);
+    float maxAperture = pow(10, depthScale);
+    aperture_.set(aperture_.get(), std::min(minAperture, aperture_.get()),
```

Comment:
Since you set many properties here you want to lock the network using NetworkLock